app/models/entities/person_and_organization.rb

```diff
@@ -53,26 +56,33 @@ def connec_model_to_external_model(connec_hash_of_entities)
   #             }
   #           }
   def external_model_to_connec_model(external_hash_of_entities)
-    contacts = external_hash_of_entities['Contact'] ? external_hash_of_entities['Contact'] : []
-    leads = external_hash_of_entities['Lead'] ? external_hash_of_entities['Lead'] : []
+    contacts = external_hash_of_entities['Contact']
+    leads = external_hash_of_entities['Lead']
 
-    modelled_hash = {'Contact' => { 'Person' => [], 'Organization' => [] }}
+    modelled_hash = {'Contact' => { 'Person' => [], 'Organization' => [] },
+                     'Lead' => {'Person' => [], 'Organization' => []}
+                    }
 
-    contacts.each do |contact|
-      if contact['is_organization']
-        modelled_hash['Contact']['Organization'] << contact
-      else
-        modelled_hash['Contact']['Person'] << contact
-      end
+    filter_entities_into_connec(modelled_hash, 'Contact', contacts, 'is_organization')
+    filter_entities_into_connec(modelled_hash, 'Lead', leads, 'organization_name', 'last_name')
+
+    modelled_hash
+  end
+
+  private
+  def split_leads_from_connec(connec_hash, leads_array, entity_array)
+    connec_hash.each do |entity|
+      entity['is_lead'] ? leads_array << entity : entity_array << entity
```

Comment:
entity_array is not the best name you can find ;) and you're not consistent leadS_array (plural) entity_array (singular)